### ISSUE TITLE
"skip" iso_checksum_type

### ISSUE THREAD

**Issue by vtolstov** [mitchellh/packer#3128]
Sometimes you have already downloaded iso and does not need to do checksum every run.
I want to define null/skip or something like this checksum type (and implement hash.Hash inside common/)
so this speedup booting.
@cbednarski @rickard-von-essen  @mitchellh what do you think?

**Comment by cbednarski** [comment]
I favor correctness over speed, personally. I think we can learn a lesson from mongoldb's infamous benchmarks here.

How much time does this actually save you? sha256 on an ubuntu iso takes <4 seconds. md5 <2 seconds. My entire build takes ~15 minutes so 4s is a drop in the bucket.

**Comment by vtolstov** [comment]
what is iso image size ? for 4Gb this takes more time...

**Comment by cbednarski** [comment]
The ISO is a bit over 500mb, so 4gb would take around 30 seconds?

**Comment by vtolstov** [comment]
yes, and this time you need on each packer run. And this highly depends on hdd speed, sometimes i'm lost is about 1m. Yes this is not so big, but if i'm already download iso, why packer do checksum on each run? why not only after first download?

**Comment by cbednarski** [comment]
The reason I'm hesitant to do this is because including the checksum before each build is a really useful way to guard against all sorts of unexpected problems. The failure mode for a checksum is pretty clear "this file is wrong".

By contrast the failure modes for a truncated source image are numerous and difficult to troubleshoot. Maybe the hypervisor fails to start. Maybe ovf tool throws an exception. Maybe there is disk corruption or IO panic trying to read the file.

There isn't really any continuity between runs so packer doesn't know if it has validated the checksum already.

**Comment by cbednarski** [comment]
@vtolstov Let's do this behind a `PACKER_SKIP_VALIDATION` environment variable. I think default packer behavior should be to validate always, but we can make this work like a debug flag for expert users like yourself who have specialized use-cases.

**Comment by dragon788** [comment]
@cbednarski Is the current Packer behavior to validate the local checksum and then download if it is incorrect or only if the local cached ISO doesn't exist? I'm frequently using files that have the same name but a different checksum and updating the checksum in the file is easy (I can do it remotely) but when I attempt to run Packer with the new checksum, it appears to be looking at the old file and bombing because the checksum is invalid rather than overwriting the file that is incorrect.

**Comment by cbednarski** [comment]
> it appears to be looking at the old file and bombing because the checksum is invalid rather than overwriting the file that is incorrect.

This code path handles [local and remote files differently](https://github.com/mitchellh/packer/blob/master/common/download.go#L101-L170). However, this behavior is not related to this issue; if you think there is a bug there please open a new issue.

**Comment by imduffy15** [comment]
Any movement on this? Would be really handy when using QEMU with an image as source, its taken as an ISO so requires a hash to be calculated (different to other image as source builders)

**Comment by vtolstov** [comment]
I'm busy with qmp support to send keys, after that i can take this.

**Comment by mwhooker** [comment]
you should be able to use `"none"` now for iso_checksum_type
